### DISCUSSION TITLE
Aeotec Picco Shutter ZGA004 limited functionality / Writing attributes with dev console

### DISCUSSION THREAD

**Discussion by Kaktus317** [Koenkk/zigbee2mqtt#24060] [Devices]
Hi,

I have some issues with the picco shutter from Aeotec. It is advertised to work with shutter blinds but apparently the standard configuration is roller shades and pressing 6x S2 does not change the configuration and starting the calibration of the shutter tilt as mentioned in the manual.Even worse: there does not seem to be a way to change this configuration via action button on the device or with the connected switch.

I also want to change the switch type to "momentary" and continues close/open mode. For the latter the manual says holding S1 and S2 for 5s. But shutter switches do not allow to press both switches at the same time.

[The manual apparently is very detailed.](https://aeotec.freshdesk.com/helpdesk/attachments/6200417572) Still I am not up for the task. I thought I could just look up the cluster/attributes and write the necessary configuration via the dev console in zigbee2mqtt. But I am stuck and in the [homeassistant communtiy an helpful user came up with an external converter](https://community.home-assistant.io/t/dev-console-in-z2m/772409) (the code does not work for me though). In my opinion having those settings on the device page in z2mqtt probably is not a bad idea though. Hence, I was hoping somebody here could help me with that?

**Comment by Kaktus317** [comment]
Hi,
I did not expect that this will be complicated. I though I could quickly set/write a cluster and just don't know how to do that.

The good: I figured out the issue and it can be done without it. The bad: It still does not work.

What do I mean? Apparently the device integration to Zigbee2Mqtt has some flaws to it. The "settings (specific) tab has the option "Invert cover: Inverts the cover position, false: open=100,close=0, true: open=0,close=100 (default false)." For me this did not work. _EDIT2: At least I thought it did not, because it does not invert up/down as well, as I thought it would._
When clicking "up" the shutter still went down. Now, by changing the cables to the device itself the open/close was correct and the tilt configuration via 6x S2 only works when the shutter is closed (which before the device thought it was not due to the wrong setting). When I do the calibration the shutter behaves differently now: Going to position X% results in going down and tilting the slats to open (which I don't want). Also I can read Endpoint 1 Cluster "closures" Attribute "windowcoveringtype" in the dev console and it's 6 instead of 0 now, which indicates that the device is in operating mode 1 (shutter mode) now. Nice. The shutter does not react to any changes in the tilt settings though. I can move from 0 to 100% and nothing happens.

Thus, I am stuck again and this time I do not even have a clue where to start looking for a solution. Maybe best just returning the devices before spending even more hours just getting them to work...

EDIT: Oh and momentary switch and continous mode just work now. No idea how that happened...

**Comment by TheAndi** [comment]
@Kaktus317 Did you get it working in the end? Thinking about getting those Aeotec Picco Shutters, too. 
But I defnitly want the tilt to work 🙈

**Comment by Kaktus317** [comment]
No, I did not.
I briefly looked into how to built an external converter. But following the guide for "unsupported devices" did not help at all, because there were no logs whatsoever when pressing tilt.

I contacted Aeotec and the support said they will look into it, but I have not heard back from them. I told them that Z2MQTT and Homeassistant support would be good marketing as well. But I do not have big hope. Which is a pity, because I think due to the very detailed manual a skilled coder should be able to implement this in no time.

And also: It seems like there are not many alternatives out there, which support tilt?! I only came across Insta Nexentro and Aeotec for Zigbee and the new Shelly 2PM Gen3 with Wifi (not tested by me). I would really like to stick with Zigbee, though. And having an automatic calibration which can be triggered via switch is a nice feature of Aeotec comparted to the Insta ones. Basically just the tilt (and current position) is not working and it would be great to have some of the settings just directly in Z2M (like switch type, invert up/down, ...).

I really hope that somebody can help us out in updating the device in Z2M or writing an external converter for it. I have to look into it again, but do not have the time atm.

**Comment by TheAndi** [comment]
Thanks for the quick reply. Very sad, given how detailed the documentation is. 

Maybe I can help you out with another alternative: NodOn Zigbee Roller Shutter (cheapest [here](https://smaart.haus/products/nodon-zigbee-rollladenschalter-sin-4-rs-20)). They support native tilting and the [calibration](https://www.zigbee2mqtt.io/devices/SIN-4-RS-20.html#calibration-rotation-run-time-up-numeric) can even be set in Z2M 🤯🥳  . 

Still wouldn't mind using Aeotec, being cheaper and engineered in Hamburg, but I am not looking forward to having a lot of trouble with it :/

**Reply by Kaktus317** [reply]
Thanks, I saw this device and somehow discarded the idea of switching to it (maybe because I was not certain that it supports tilt?) Anyways: I bought three of them and they work fine. Thanks for pointing them out again for me :)

**Comment by c0ldtech** [comment]
Tagging along here, since I had issues with the ZGA004 and now switched to Nodon.

Unfortunately, both don't work well for me. I was able to get the Aeotec to work including tilt ONLY in homeassistant, but there it worked perfectly. In zigbee2mqtt I never managed to make tilt work. It reported tilt position, but trying to set tilt position did not do anything. I might have just switched to homeassistant, but unfortunately I have some sensors that are not supported (yet?) by HA, so that's a no-go.
Also, the Aeotec has a big lag between using the physical switches and it actually starting/stopping to move. That makes manually adjusting tilt nearly impossible.

Now on the Nodon: for me it has a very erratic behaviour regarding tilt. Yes, tilt does work, but I can never get the adjustment right. For example I'll send a tilt value of 50%, and it will tilt, but a bit too much - and at the same time, it will not report 50%, but anything above or below.
Then trying to set 55% to get the tilt angle that I want, it will overshoot way too much (and again report a completely different tilt).

Also, I can not get scenes to work. I'll create a scene with position 80% and tilt 50%, but when trying to recall the scene, the shutter will only move for a little bit and then stop. It seems like it does try to do position and tilt at the same time instead of waiting for position to finish and then doing tilt.

Do you guys not have any of these issues?
